catroid/src/test/java/org/catrobat/catroid/test/web/WebConnectionHolderTest.java

```diff
@@ -0,0 +1,132 @@
+
+		isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(5, webConnectionHolder.getConnections().size());
+
+		isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(6, webConnectionHolder.getConnections().size());
+
+		isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(7, webConnectionHolder.getConnections().size());
+
+		isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(8, webConnectionHolder.getConnections().size());
+
+		isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(9, webConnectionHolder.getConnections().size());
+
+		isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(10, webConnectionHolder.getConnections().size());
+
+		isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertFalse(isConnectionSuccessfullyAddedToList);
+		assertEquals(10, webConnectionHolder.getConnections().size());
+	}
+
+	@Test
+	public void testRemoveConnectionFromList() {
+		WebConnection webConnection = Mockito.mock(WebConnection.class);
+		WebConnectionHolder webConnectionHolder = new WebConnectionHolder();
+		boolean isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(1, webConnectionHolder.getConnections().size());
+
+		webConnectionHolder.removeConnectionFromList(webConnection);
+		assertEquals(0, webConnectionHolder.getConnections().size());
+	}
+
+	@Test
+	public void testWebConnectionHolderOnPause() {
+		WebConnection webConnection = Mockito.mock(WebConnection.class);
+		WebConnectionHolder webConnectionHolder = new WebConnectionHolder();
+
+		boolean isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(1, webConnectionHolder.getConnections().size());
+
+		isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(2, webConnectionHolder.getConnections().size());
+
+		isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(3, webConnectionHolder.getConnections().size());
+
+		webConnectionHolder.onPause();
```

Comment:
also check that cancelCall is called on every connection. (and test for one connection is probably sufficient and easier to read and comprehend)